### ISSUE TITLE
pip wheel breaks version number

### ISSUE THREAD

**Issue by bennuttall** [nmichlo/tonic-config#1]
Due to the calculation of version number from the git tree in setup.py, the command `pip wheel` has unexpected results:

```bash
$ pip wheel tonic-config --no-deps
```

```
tonic_config-fatal_.not.a.git.repository._or.any.of.the.parent.directories_..git.devfatal_.not.a.git.repository._or.any.of.the.parent.directories_..git_fatal_.not.a.git.repository._or.any.of.the.parent.directories_..git.fatal_.n_dirty-py3-none-any.whl
```

I run a service that creates wheels of all packages on PyPI, and the length of the filename broke it 😱 

I'd suggest you catch the "not a git repository" error and fall back to the normal version number. Do the same for the case where git is not installed.

**Comment by nmichlo** [comment]
This should now be resolved as of 5d9bcf4165c452a166e0cae2865c223d756b44fb?

**Comment by bennuttall** [comment]
Yep great thanks 👍
